pallets/evm-accounts/src/lib.rs

```diff
@@ -314,10 +366,107 @@ pub mod pallet {
 			Self::deposit_event(Event::ContractDisapproved { address });
 			Ok(())
 		}
+
+		/// Proves ownership of an account and binds it to the EVM address.
+		/// This is useful for accounts that want to submit some substrate transaction, but only
+		/// received some ERC20 balance and `System` pallet doesn't register them as a substrate account.
+		///
+		/// Parameters:
+		/// - `origin`: Unsigned origin.
+		/// - `account`: Account proving ownership of the address.
+		/// - `asset_id`: Asset ID to be set as fee currency for the account.
+		/// - `signature`: Signed message by the account that proves ownership of the account.
+		///
+		/// Emits `AccountClaimed` event when successful.
+		#[pallet::call_index(6)]
+		#[pallet::weight(<T as Config>::WeightInfo::claim_account())]
+		pub fn claim_account(
+			origin: OriginFor<T>,
+			account: T::AccountId,
+			asset_id: T::AssetId,
+			signature: Signature,
+		) -> DispatchResult {
+			ensure_none(origin)?;
+
+			Self::verify_claim_account(&account, asset_id, signature)?;
+
+			Self::do_bind_evm_address(&account)?;
+
+			T::FeeCurrency::set(&account, asset_id)?;
```

Comment:
i am wondering if we should let it succeed even if this fails. 

and it is actually a little concern. 

reasoning is that there might be a situation where an asset that you have cannot be used as fee payment asset, but you stil want to bind and claim the account. 
this would prevent it - and you could just actually spam it forever for free, as it would pass the validate_unsigned but fails here on execution